src/main/java/org/bytedeco/javacpp/tools/Generator.java

```diff
@@ -293,6 +294,11 @@ boolean classes(boolean handleExceptions, boolean defineAdapters, boolean conver
         out.println("    typedef int jint;");
         out.println("    typedef long long jlong;");
         out.println("    typedef signed char jbyte;");
+        out.println("    #if __cplusplus >= 201103L || _MSC_VER >= 1900");
+        out.println("        #define THREADLOCAL thread_local");     
+        out.println("    #else");
+        out.println("        #define THREADLOCAL __declspec(thread)");     
+        out.println("    #endif");        
```

Comment:
I'd rather not define a new symbol since we're only using it in a single place.

Reply:
I will change it if you like your way better. Personally, I find the nested macro difficult to read.

Reply:
This is also a nested macro, isn't? We can make the other one a bit better, for example, I think this works:

```patch
--- a/src/main/java/org/bytedeco/javacpp/tools/Generator.java
+++ b/src/main/java/org/bytedeco/javacpp/tools/Generator.java
@@ -525,7 +525,12 @@ public class Generator {
         out.println("}");
         out.println();
         out.println("#if !defined(NO_JNI_DETACH_THREAD) && defined(_WIN32)");
-        out.println("   static __declspec(thread) struct JavaCPP_thread_local {");
+        out.println("#if __cplusplus >= 201103L || _MSC_VER >= 1900");
+        out.println("   thread_local");
+        out.println("#else");
+        out.println("   __declspec(thread)");
+        out.println("#endif");
+        out.println("   static struct JavaCPP_thread_local {");
         out.println("       JNIEnv* env = NULL;");
         out.println("       ~JavaCPP_thread_local() {");
         out.println("           if (env && JavaCPP_vm) {");
```

Reply:
Does not compile. Looks like it needs to come before static.

Reply:
Ok, so this works for sure and doesn't look bad:
```patch
--- a/src/main/java/org/bytedeco/javacpp/tools/Generator.java
+++ b/src/main/java/org/bytedeco/javacpp/tools/Generator.java
@@ -525,7 +525,12 @@ public class Generator {
         out.println("}");
         out.println();
         out.println("#if !defined(NO_JNI_DETACH_THREAD) && defined(_WIN32)");
-        out.println("   static __declspec(thread) struct JavaCPP_thread_local {");
+        out.println("#if __cplusplus >= 201103L || _MSC_VER >= 1900");
+        out.println("   static thread_local");
+        out.println("#else");
+        out.println("   static __declspec(thread)");
+        out.println("#endif");
+        out.println("   struct JavaCPP_thread_local {");
         out.println("       JNIEnv* env = NULL;");
         out.println("       ~JavaCPP_thread_local() {");
         out.println("           if (env && JavaCPP_vm) {");
```